pkg/kubelet/images/image_manager.go

```diff
@@ -136,6 +138,19 @@ func (m *imageManager) EnsureImageExists(pod *v1.Pod, container *v1.Container, p
 			msg := fmt.Sprintf("image pull failed for %s because the registry is unavailable.", container.Image)
 			return "", msg, imagePullResult.err
 		}
+		// Pull image failure because image is corrupt: https://github.com/kubernetes/kubernetes/issues/46184
+		if strings.Contains(imagePullResult.err.Error(), "rename") && strings.Contains(imagePullResult.err.Error(), "directory not empty") {
```

Comment:
The other idea is to start a PR that moves the code base to the currently supported engine-api (https://github.com/moby/moby/tree/master/client) which should also give 1.13.X and higher support.